### PR TITLE
Use ScreenCoordsXY for scroll_mouse* functions

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -561,12 +561,12 @@ static void window_map_scrollgetsize(rct_window* w, int32_t scrollIndex, int32_t
...
  */
-static void window_map_scrollmousedown(rct_window* w, int32_t scrollIndex, int32_t x, int32_t y)
+static void window_map_scrollmousedown(rct_window* w, int32_t scrollIndex, ScreenCoordsXY screenCoords)
 {
-    CoordsXY c = map_window_screen_to_map(x, y);
+    CoordsXY c = map_window_screen_to_map(screenCoords);
     int32_t mapX = std::clamp(c.x, 0, MAXIMUM_MAP_SIZE_TECHNICAL * 32 - 1);
     int32_t mapY = std::clamp(c.y, 0, MAXIMUM_MAP_SIZE_TECHNICAL * 32 - 1);
-    int32_t mapZ = tile_element_height({ x, y });
+    int32_t mapZ = tile_element_height({ screenCoords.x, screenCoords.y });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Map.cpp]: This is mixing up `ScreenCoordsXY` and `CoordsXY` which you might want to chime in @duncanspumpkin

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Map.cpp]: Yes looks like a mistake should be mapx and mapy

Comment (tupaschoal) [inline src/openrct2-ui/windows/Map.cpp]:
```suggestion
    int32_t mapZ = tile_element_height({ mapX, mapY });
```

Comment (tupaschoal) [reply src/openrct2-ui/windows/Map.cpp]: Done!

**Pull request merged**